great_expectations/core/factory/checkpoint_factory.py

```diff
@@ -41,7 +45,15 @@ def add(self, checkpoint: Checkpoint) -> Checkpoint:
         self._store.add(key=key, value=checkpoint)
 
         # TODO: Add id adding logic to CheckpointStore to prevent round trip
-        return self._get(key=key)
+        obj = self._get(key=key)
+
+        submit_event(
+            event=CheckpointCreatedEvent(
+                checkpoint_id=obj.id,
+            )
+        )
```

Comment:
This currently round trips so the actual object is different - a subtle difference from other stores (that might directly add the ID to the input object)